src/rust/derive-dynamic-node/src/lib.rs

```diff
@@ -15,16 +15,35 @@ use syn::{parse_quote,
           NestedMeta,
           Type};
 
-const CREATED_TIME: &'static str = "created_time";
+const CREATE_TIME: &'static str = "create_time";
 const LAST_SEEN_TIME: &'static str = "last_seen_time";
-const TERMINATED_TIME: &'static str = "terminated_time";
+const TERMINATE_TIME: &'static str = "terminate_time";
+const IMMUTABLE: &'static str = "immutable";
+const INCREMENT: &'static str = "increment";
+const DECREMENT: &'static str = "decrement";
 
-fn name_and_ty(field: &Field) -> (&Ident, &Type) {
-    (field.ident.as_ref().unwrap(), &field.ty)
+fn name_and_ty(field: &Field) -> (&Ident, &Type, String) {
+    let mut resolution = None;
+    for attr in &field.attrs {
+        on_grapl_attrs(&attr, |attr| {
+            match attr {
+                IMMUTABLE => resolution = Some(attr.to_string()),
+                INCREMENT => resolution = Some(attr.to_string()),
+                DECREMENT => resolution = Some(attr.to_string()),
```

Comment:
You could probably just use one match arm on `IMMUTABLE | INCREMENT | DECREMENT`.